### PR TITLE
Share Dashboard Feedback

### PR DESCRIPTION
**Pull request by lovincyrus**
https://rilldata.slack.com/archives/C07PRPUFCE7/p1730160179703019

https://www.figma.com/design/Qt6EyotCBS3V6O31jVhMQ7/RILL-Latest?node-id=17077-830426&node-type=frame&t=VQugAVqX0LGPrjVJ-0

- [x] Disable role change for invited users
- [x] https://rilldata.slack.com/archives/C07PRPUFCE7/p1730426134345309?thread_ts=1730160179.703019&cid=C07PRPUFCE7 
- [x] Comment out unimplemented changes until groups management is implemented

CleanShot 2024-10-29 at 11 05 20@2x

### CODE REVIEW

**Review by ericpgreen2 — COMMENTED** [review]
The Organization/Project section in the Share Dashboard popover looks different from the Figma mocks:

image

Though, in looking at these mocks, I'm not clear about the "manage" links. Do both links go to the same place (the birds-eye-view users table) or different places? CC @jkhwu

**Comment by lovincyrus** [comment]
> The Organization/Project section in the Share Dashboard popover looks different from the Figma mocks:
> 
> image
> Though, in looking at these mocks, I'm not clear about the "manage" links. Do both links go to the same place (the birds-eye-view users table) or different places? CC @jkhwu

we can move the convo to https://rilldata.slack.com/archives/C07PRPUFCE7/p1730160179703019

**Comment by jkhwu** [comment]
Replied in Slack

**Comment by lovincyrus** [comment]
The main change scope of this PR is to disallow role change for pending users. I've also commented out other WIP changes (e.g. click on 'Manage' button in the share project popover to manage users within the project). We can revisit this when group management is implemented. @ericpgreen2 

Ready for +1 and 🚢

**Comment by lovincyrus** [comment]
Handled https://rilldata.slack.com/archives/D07ED3LUUES/p1731607880555669 @ericokuma 

CleanShot 2024-11-14 at 10 49 29@2x

File: web-admin/src/features/projects/user-management/UsergroupItem.svelte [hunk 1]
```diff
@@ -35,7 +36,9 @@
     on:blur={() => (isHovered = false)}
   >
     <UsergroupAvatar {organization} usergroup={group.groupName} />
-    <UsergroupSetRole {organization} {project} {group} />
+    {#if !canManage}
```
Comment (ericpgreen2) [inline web-admin/src/features/projects/user-management/UsergroupItem.svelte:39]: Is this reversed? `canManage` should be able to set role, right?

File: web-admin/src/features/projects/user-management/OrganizationItem.svelte [hunk 2]
```diff
@@ -41,9 +46,12 @@
         shape="square"
         name={organization}
         count={userGroupMemberUsersCount}
-        isEveryFromText
+        isEveryoneFromText
+        canManage={onMetricsExplorerPage}
       />
-      <UserManagementOrganizationSetRole {organization} {project} {group} />
+      {#if !canManage}
```
Comment (ericpgreen2) [inline web-admin/src/features/projects/user-management/OrganizationItem.svelte:52]: Likewise, is this reversed too, and it should be `canManage`?

File: web-admin/src/features/projects/user-management/OrganizationItem.svelte [hunk 3]
```diff
@@ -41,9 +46,12 @@
         shape="square"
         name={organization}
         count={userGroupMemberUsersCount}
-        isEveryFromText
+        isEveryoneFromText
+        canManage={onMetricsExplorerPage}
```
Comment (ericpgreen2) [inline web-admin/src/features/projects/user-management/OrganizationItem.svelte:50]: I'm a little confused how this `canManage` is different from the prop `canManage`

**Review by ericpgreen2 — APPROVED** [review]

File: web-admin/src/features/projects/user-management/UsergroupItem.svelte (hunk 1 above)
Comment (lovincyrus) [reply web-admin/src/features/projects/user-management/UsergroupItem.svelte:39]: `canManage` is to trigger the user management dialog when manage text is clicked.

Comment (lovincyrus) [reply web-admin/src/features/projects/user-management/UsergroupItem.svelte:39]: the feature is commented out right now

**Comment by lovincyrus** [comment]
Removing the `canManage` WIP code @ericpgreen2

**Pull request merged**